src/Microsoft.Health.Fhir.TemplateManagement/Factory/TemplateCollectionProviderFactory.cs

```diff
@@ -59,19 +56,22 @@ public ITemplateCollectionProvider CreateTemplateCollectionProvider(string image
             return new TemplateCollectionProvider(imageInfo, client, _templateCache, _configuration);
         }
 
-        public void InitDefaultTemplates()
+        private void InitDefaultTemplates()
         {
-            TemplateLayer defaultTemplateLayer = TemplateLayer.ReadFromEmbeddedResource();
-            _templateCache.Set(ImageInfo.DefaultTemplateImageReference, defaultTemplateLayer, new MemoryCacheEntryOptions() { AbsoluteExpiration = ObjectCache.InfiniteAbsoluteExpiration, Size = defaultTemplateLayer.Size, Priority = Extensions.Caching.Memory.CacheItemPriority.NeverRemove });
-            _templateCache.Set(defaultTemplateLayer.Digest, defaultTemplateLayer, new MemoryCacheEntryOptions() { AbsoluteExpiration = ObjectCache.InfiniteAbsoluteExpiration, Size = defaultTemplateLayer.Size, Priority = Extensions.Caching.Memory.CacheItemPriority.NeverRemove });
+            var memoryOption = new MemoryCacheEntryOptions() { AbsoluteExpiration = ObjectCache.InfiniteAbsoluteExpiration, Priority = Extensions.Caching.Memory.CacheItemPriority.NeverRemove };
+            foreach (var templateInfo in DefaultTemplateInfo.DefaultTemplateMap)
+            {
+                TemplateLayer templateLayer = TemplateLayer.ReadFromEmbeddedResource(templateInfo.Value.TemplatePath);
+                memoryOption.SetSize(templateLayer.Size);
+                _templateCache.Set(templateInfo.Value.ImageReference, templateLayer, memoryOption);
+                _templateCache.Set(templateLayer.Digest, templateLayer, memoryOption);
+            }
         }
 
-        public void InitDefaultTemplates(string path)
+        public void InitDefaultTemplates(DefaultTemplateInfo templateInfo)
         {
-            path ??= Path.Combine(AppDomain.CurrentDomain.BaseDirectory, Constants.DefaultTemplatePath);
-
-            TemplateLayer defaultTemplateLayer = TemplateLayer.ReadFromFile(path);
-            _templateCache.Set(ImageInfo.DefaultTemplateImageReference, defaultTemplateLayer, new MemoryCacheEntryOptions() { AbsoluteExpiration = ObjectCache.InfiniteAbsoluteExpiration, Size = defaultTemplateLayer.Size, Priority = Extensions.Caching.Memory.CacheItemPriority.NeverRemove });
+            TemplateLayer defaultTemplateLayer = TemplateLayer.ReadFromFile(templateInfo.TemplatePath);
+            _templateCache.Set(templateInfo.ImageReference, defaultTemplateLayer, new MemoryCacheEntryOptions() { AbsoluteExpiration = ObjectCache.InfiniteAbsoluteExpiration, Size = defaultTemplateLayer.Size, Priority = Extensions.Caching.Memory.CacheItemPriority.NeverRemove });
```

Comment:
Can we extract MemoryCacheEntryOptions out, like the one we do for `InitDefaultTemplates()`?

Reply:
sure, done.